packages/backend-common/src/reading/BitbucketUrlReader.ts

```diff
@@ -125,14 +121,31 @@ export class BitbucketUrlReader implements UrlReader {
       throw new Error(message);
     }
 
-    let folderPath = `${project}-${repoName}`;
-    if (isHosted) {
-      folderPath = `${project}-${repoName}-${lastCommitShortHash}`;
```

Comment:
This was incorrect. It should have been something similar to `my-repo-1-master@d373ed91095` as @kaparwan points out in #4121

Anyway, the API response header is responsible for telling us the filename they are sending over.